sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,215 @@
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| cleanup()            | Method is called when planner server goes to on_cleanup state. Ideally this| Yes                     |
+|                      | method should clean up resoures which are created for the planner.         |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+| createPlan()         | Method is called when planner server demands a global plan for specified   | Yes                     |
+|                      | start and goal pose. This method returns `nav_msgs::msg::Path` carrying    |                         |
+|                      | global plan. This method takes 2 input parmas: start pose and goal pose.   |                         |
++----------------------+----------------------------------------------------------------------------+-------------------------+
+
+For this tutorial, we will be using methods ``StraightLine::configure()`` and ``StraightLine::createPlan()`` to create straight line planner.
+
+In our planner, ``StraightLine::configure()`` method sets member variables and ROS parameters,
+
+.. code-block:: c++
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(node_, name_ + ".interpolation_resolution", rclcpp::ParameterValue(0.1));
+  node_->get_parameter(name_ + ".interpolation_resolution", interpolation_resolution_);
+
+In ``StraightLine::createPlan()`` method, straight line planner is called using start pose and goal pose to solve the global path planning problem. Upon succeding, it converts the path to the ``nav_msgs::msg::Path`` and returns to the caller which is planner server instance. Below annotation shows the implementation of this method.
+
+.. code-block:: c++
+
+  nav_msgs::msg::Path global_path;
+
+  // Checking if the goal and start state is in the global frame
+  if (start.header.frame_id != global_frame_) {
+    RCLCPP_ERROR(
+      node_->get_logger(), "Planner will only except start position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  if (goal.header.frame_id != global_frame_) {
+    RCLCPP_INFO(
+      node_->get_logger(), "Planner will only except goal position from %s frame",
+      global_frame_.c_str());
+    return global_path;
+  }
+
+  global_path.poses.clear();
+  global_path.header.stamp = node_->now();
+  global_path.header.frame_id = global_frame_;
+  global_path.poses.push_back(start);
+  // calculating the number of loops for current value of interpolation_resolution_
+  int total_number_of_loop = std::abs(goal.pose.position.x - start.pose.position.x)/interpolation_resolution_;
```

Comment:
This should probably be total distance, e.g. (`hypot(delta x, delta y)`) and then interpolating along the total distance, this would have very different results if the path was in a straight line along the Y axis (e.g. number of loops = 0).